pkg/operation/botanist/wanted_secrets.go

```diff
@@ -481,6 +481,13 @@ func (b *Botanist) generateWantedSecretConfigs(basicAuthAPIServer *secrets.Basic
 			UsedForSSH: true,
 		},
 
+		// Secret definition for ssh-keypair.old
+		&secrets.RSASecretConfig{
+			Name:       v1beta1constants.SecretNameOldSSHKeyPair,
+			Bits:       4096,
+			UsedForSSH: true,
+		},
```

Comment:
For my understanding, why is this here?

Reply:
Without this, the SecretManager never considered `ssh-keypair.old` to be a secret that it should sync back up to the garden. And I was under the impression that both SSH keys should be on the garden. It doesn't make sense to have a RSAConfig here, because I do not want a new key to be generated from scratch, but it was the only way I found to make both keys appear on the garden.

Reply:
OK, thanks. Fine for me for now. cc @plkokanov as initial contributor of this secrets manager: I think we should simplify this soon because it's somewhat confusing and complex

Reply:
Yeah if we keep this, there needs to be a comment on this explaining why it's there.

Reply:
To further clarify: With the current code, each new shoot gets 2 SSH keypairs immediately, even before the first rotation. It ain't pretty, but that's how far I got until now. ^^ I was hoping we can consider this a ... happy little accident. 🎨

Reply:
Sorry for commenting so late, the flow for the secrets generation currently is the following:
In the GenerateSecrets step it will get all secret configs defined in `generateWantedSecretConfigs` generate the secret metadata for each and save that in the `ShootState`. 
Then in the DeploySecrets step it will again get all required secrets from `generateWantedSecretConfigs` and for each secret configuration it will get the corresponding metadata from the `ShootState`, combine it with the static data that comes in the secret config and deploy a `Secret` object in the shoot's control plane.

I can spend some time to see if there is a way to avoid having both the old and new ssh configs added in `generateWantedSecretConfigs`. 

As for the whole secret management I have a couple of ideas on how to simplify it and add support for rotating credentials that would require keeping the old secret (e.g. etcd encryption which is currently done in separate  steps)